### PR TITLE
add rehype-slug and rehype-prism to website

### PR DESCRIPTION
**Pull request by BarryThePenguin**
> [I highly encourage changing this.](https://github.com/keystonejs/keystone-5/blob/a15601ee6b551eae04f41289c339dbef8315843a/website/src/components/markdown/Code.js#L15)

- Adds [`rehype-slug`](https://github.com/rehypejs/rehype-slug) for consistent heading ids

- Adds [`@mapbox/rehype-prism`](https://github.com/mapbox/rehype-prism) to highlight code blocks

- Removes the need for `prism-react-renderer` and `react-addons-text-content`

- Updates `gatsby-mdx` to v0.6 which introduces `mdx` v1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -117,7 +118,7 @@
     "fixturez": "^1.1.0",
     "flushable": "^1.0.0",
     "fs-extra": "^7.0.0",
-    "gatsby-mdx": "^0.4.4",
+    "gatsby-mdx": "^0.6.3",
```
Comment (BarryThePenguin) [inline package.json]: This updates `mdx` to v1. This PR takes care of the breaking changes mentioned in [their migration guide](https://mdxjs.com/migrating/v1)

File: website/src/components/markdown/Code.js [hunk 2]
```diff
@@ -1,46 +1,41 @@
...
-import React from 'react'; // eslint-disable-line
-import Highlight, { defaultProps } from 'prism-react-renderer';
-import reactAddonsTextContent from 'react-addons-text-content';
 import { jsx } from '@emotion/core';
 
 import theme from '../../prism-themes/custom';
 
-/*
-This solution should be seen as extremely temporary and not a platonic ideal of code highlighting.
+function themeToDict(theme, language) {
```
Comment (BarryThePenguin) [inline website/src/components/markdown/Code.js]: This is a slight modification of the [`themeToDict()` utilitly from `prism-react-renderer`](https://github.com/FormidableLabs/prism-react-renderer/blob/master/src/utils/themeToDict.js)

File: website/src/components/markdown/Heading.js [hunk 3]
```diff
@@ -1,21 +1,13 @@
...
 import React from 'react'; // eslint-disable-line no-unused-vars
-import reactAddonsTextContent from 'react-addons-text-content';
-import snakeCase from 'lodash.snakecase';
 import { jsx } from '@emotion/core';
 import { colors, gridSize } from '@arch-ui/theme';
 import { LinkIcon } from '@arch-ui/icons';
 
 import { mq } from '../../utils/media';
 
-function dashcase(children) {
```
Comment (BarryThePenguin) [inline website/src/components/markdown/Heading.js]: `rehype-slug` removes the need for custom slug generation

File: website/src/templates/docs.js [hunk 4]
```diff
@@ -303,7 +303,7 @@ const Content = props => (
         boxShadow: '-4px 0 0 rgba(9, 30, 66, 0.09)',
         boxSizing: 'border-box',
         fontFamily: 'Consolas,Menlo,Monaco,"Andale Mono","Ubuntu Mono",monospace',
-        lineHeight: '1.4',
+        lineHeight: '1.2',
```
Comment (BarryThePenguin) [inline website/src/templates/docs.js]: Looks like the difference is that `prism-react-renderer` uses block elements for lines of code but `@mapbox/rehype-prism` uses inline elements

The lineHeight change can be reverted, I changed it so this PR would be visually similar to the current site

**Comment by timleslie** [comment]
Thanks @BarryThePenguin! @Noviny could you please have a look over this and approve if you're happy with it.

**Comment by jesstelford** [comment]
Might this also fix https://github.com/keystonejs/keystone-5/issues/1271 ?

**Comment by BarryThePenguin** [comment]
> Might this also fix #1271 ?

Should do 👍 Adding `rehype-slug` aligns the TOC links with the generated slugs

**Pull request merged**